edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/Interfaces/IMetricsScraper.cs

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    public interface IMetricsScraper
+    {
+        Task<IEnumerable<Metric>> ScrapeAsync(CancellationToken cancellationToken);
```

Comment:
Please read about this, https://stackoverflow.com/questions/23295119/asynchronous-iterator-taskienumerablet.  That's why I suggest to use Task<Stream>.

Reply:
did you think about using IAsyncEnumerable? the above link has an example to read a file with IAsyncEnumerable return type.

Reply:
IAsyncEnumerable is not in .net standard 2.0